api_shield_operations.go

```diff
@@ -0,0 +1,221 @@
+	OrderBy string
+	// Filters to only return operations that match filtering criteria, see APIShieldGetOperationsFilters
+	Filters *APIShieldGetOperationsFilters
+	// Pagination options to apply to the request.
+	Pagination *PaginationOptions
+}
+
+// APIShieldGetOperationsFilters represents the filtering query parameters to set when retrieving operations
+//
+// See API documentation https://developers.cloudflare.com/api/operations/api-shield-endpoint-management-retrieve-information-about-all-operations-on-a-zone
+type APIShieldGetOperationsFilters struct {
+	// Host filters results to only include the specified hosts.
+	Hosts []string
+	// Host filters results to only include the specified methods.
+	Methods []string
+	// Endpoint filter results to only include endpoints containing this pattern.
+	Endpoint string
+}
+
+// APIShieldGetOperationResponse represents the response from the api_gateway/operations/{id} endpoint.
+type APIShieldGetOperationResponse struct {
+	Result APIShieldOperation `json:"result"`
+	Response
+}
+
+// APIShieldGetOperationsResponse represents the response from the api_gateway/operations endpoint.
+type APIShieldGetOperationsResponse struct {
+	Result     []APIShieldOperation `json:"result"`
+	ResultInfo `json:"result_info"`
+	Response
+}
+
+// APIShieldDeleteOperationResponse represents the response from the api_gateway/operations/{id} endpoint (DELETE).
+type APIShieldDeleteOperationResponse struct {
+	Result interface{} `json:"result"`
+	Response
+}
+
+// GetAPIShieldOperation returns information about an operation
+//
+// API documentation https://developers.cloudflare.com/api/operations/api-shield-endpoint-management-retrieve-information-about-an-operation
+func (api *API) GetAPIShieldOperation(ctx context.Context, rc *ResourceContainer, operationID string, params *APIShieldGetOperationParams) (*APIShieldOperation, error) {
+	uri := fmt.Sprintf("/zones/%s/api_gateway/operations/%s", rc.Identifier, operationID)
+
+	if params != nil {
+		uri = strings.Join([]string{uri, params.Encode()}, "?")
+	}
+
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var asResponse APIShieldGetOperationResponse
+	err = json.Unmarshal(res, &asResponse)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+
+	return &asResponse.Result, nil
+}
+
+// GetAPIShieldOperations retrieve information about all operations on a zone
+//
+// API documentation https://developers.cloudflare.com/api/operations/api-shield-endpoint-management-retrieve-information-about-all-operations-on-a-zone
+func (api *API) GetAPIShieldOperations(ctx context.Context, rc *ResourceContainer, params *APIShieldGetOperationsParams) ([]APIShieldOperation, ResultInfo, error) {
+	uri := fmt.Sprintf("/zones/%s/api_gateway/operations", rc.Identifier)
+	if params != nil {
+		uri = strings.Join([]string{uri, params.Encode()}, "?")
```

Comment:
instead of manually building the URL, you can use [`buildURI`](https://github.com/cloudflare/cloudflare-go/blob/f968533e2861e3d84c52e6e5f194c31038604580/utils.go#L13) and `url` struct tags to handle this automatically (and consistently).

Reply:
fixed in https://github.com/cloudflare/cloudflare-go/pull/1397/commits/583a0d227aeb5d06bbfbb9d9df4aedf0eb887219